packages/redux/src/recommendations/selectors.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @module recommendations/selectors
+ * @category Recommendations
+ * @subcategory Selectors
+ */
+
+import {
+  getAreRecommendationsLoading,
+  getRecommendations,
+  getRecommendationsError,
+} from './reducer';
+import get from 'lodash/get';
+import type { State } from './types';
+import type { StoreState } from '../types';
+
+/**
+ * Checks if product recommendations has error by strategy name.
+ *
+ * @function
+ *
+ * @param state - Application state.
+ * @param strategyName - The strategy name for the specific set of recommendations.
+ *
+ * @returns {(object | undefined)} Product recommendations error.
+ */
+export const getProductRecommendationsError = (
+  state: StoreState,
+  strategyName: string,
+) => get(getRecommendationsError(state.recommendations), strategyName);
+
+/**
+ * Checks if product recommendations are loading based on a strategy name.
+ *
+ * @function
+ *
+ * @param state - Application state.
+ * @param strategyName - The strategy name for the specific set of recommendations.
+ *
+ * @returns If the product recommendations are loading or not.
+ */
+export const isProductRecommendationLoading = (
+  state: StoreState,
+  strategyName: string,
+): boolean =>
+  get(getAreRecommendationsLoading(state.recommendations), strategyName);
+
+/**
+ * Gets all the product recommendations results.
+ *
+ * @function
+ *
+ * @param state - Application state.
+ *
+ * @returns The result object containing the product recommendations.
+ */
+export const getProductRecommendations = (
+  state: StoreState,
+): State | undefined => getRecommendations(state.recommendations);
+
+/**
+ * Gets the product recommendations for a specific strategy.
+ *
+ * @function
+ *
+ * @param  state - Application state.
+ * @param  strategyName - The strategy name for the specific set of recommendations.
+ *
+ * @returns The result object containing the product recommendations.
+ */
+export const getProductRecommendationsByStrategyName = (
+  state: StoreState,
+  strategyName: string,
+): unknown | undefined => get(getProductRecommendations(state), strategyName);
```

Comment:
In this case it is preferable to return unknown only as in practice the user will have to check the value anyway.

Reply:
Source updated, can you check again.